txircd/modules/extra/alias.py

```diff
@@ -0,0 +1,55 @@
+from twisted.plugin import IPlugin
+from txircd.config import ConfigValidationError
+from txircd.module_interface import Command, ICommand, IModuleData, ModuleData
+from zope.interface import implements
+import re
+
+validCommand = re.compile(r"^[A-Z]+$")
+
+class CommandAlias(ModuleData):
+	implements(IPlugin, IModuleData)
+
+	name = "CommandAlias"
+
+	def userCommands(self):
+		commands = []
+		for command, replacement in self.ircd.config.get("command_aliases", {}).iteritems():
+			commands.append((command, 1, UserAlias(replacement)))
+		return commands
+
+	def verifyConfig(self, config):
+		if "command_aliases" in config:
+			if not isinstance(config["command_aliases"], dict):
+				raise ConfigValidationError("command_aliases", "value must be a dictionary")
+			for command, replacement in config["command_aliases"].iteritems():
+				if not isinstance(command, basestring) or not validCommand.match(command):
+					raise ConfigValidationError("command_aliases", "alias \"{}\" is not a valid command".format(command))
+				if not isinstance(replacement, basestring):
+					raise ConfigValidationError("command_aliases", "replacement \"{}\" must be a string".format(replacement))
+				replacementParts = replacement.split(' ')
+				if not validCommand.match(replacementParts[0]):
+					raise ConfigValidationError("command_aliases", "replacement \"{}\" is not a valid command".format(replacementParts[0]))
+				if any(replacementPart.startswith(":") for replacementPart in replacementParts):
+					raise ConfigValidationError("command_aliases", "replacement \"{}\" is invalid; a final parameter and \"%p\" cannot be used in the same replacement".format(replacement))
+				if replacementParts[0] in config["command_aliases"]: # Prevent infinite recursion by disallowing aliases of aliases
+					raise ConfigValidationError("command_aliases", "replacement \"{}\" may not be used as a replacement because it is an alias".format(replacementParts[0]))
+
+class UserAlias(Command):
+	implements(ICommand)
+
+	def __init__(self, replacement):
+		self.replacement = replacement
+
+	def parseParams(self, user, params, prefix, tags):
+		return {
+			"params": params,
+			"prefix": prefix,
+			"tags": tags
+		}
+
+	def execute(self, user, data):
+		replacementParts = [x if x != '%p' else ' '.join(data["params"]) for x in self.replacement.split(' ')]
+		user.handleCommand(replacementParts[0], replacementParts[1:], data["prefix"], data["tags"])
```

Comment:
This won't work as expected if a final parameter is present in the replacement. If you have a replacement of `COMMAND param1 param2 :param 3 is here`, you'll get parameters of `["param1", "param2", ":param", "3", "is", "here"]`.

When you're splitting `replacementParts`, you should make sure to split the final parameter off correctly (first splitting on `" :"` maximum 1 should be sufficient).